app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,240 @@
+require 'azure_mgmt_compute'
+require 'azure_mgmt_resources'
+require 'azure_mgmt_network'
+require 'azure_mgmt_storage'
+
+module ForemanAzureRM
+  class AzureRMCompute
+    attr_accessor :sdk
+    attr_accessor :azure_vm
+
+    Compute = Azure::Compute::Profiles::Latest::Mgmt
+    ComputeModels = Compute::Models
+
+    def initialize(azure_vm: ComputeModels::VirtualMachine.new, sdk: sdk)
+      @azure_vm = azure_vm
+      @sdk = sdk
+    end
+
+    def id
+      @azure_vm.id
+    end
+
+    def name
+      @azure_vm.name
+    end
+
+    def persisted?
+      !!identity && !!id
+    end
+
+    def region
+      @azure_vm.location
+    end
+
+    def region=(setregion)
+      @azure_vm.location = setregion
+    end
+
+    def resource_group
+      @azure_vm.resource_group
+    end
+
+    def resource_group=(setrg)
+      @azure_vm.resource_group = setrg
+    end
+
+    def vm_size
+    end
+
+    def vm_size=(setvsize)
+      @azure_vm.hardware_profile.vm_size = setvsize
+    end
+
+    def platform
+    end
+
+    def platform=(setplfrm)
+      @azure_vm.platform = setplfrm
+    end
+
+    def username
+    end
+
+    def username=(setuname)
+      @azure_vm.username = setuname
+    end
+
+    def password
+    end
+
+    def password=(setpswd)
+      @azure_vm.password = setpswd
+    end
+
+    def ssh_key_data
+    end
+
+    def os_disk_size
+    end
+
+    def os_disk_size=(setosdisk)
+      @azure_vm.os_disk.disk_size_gb = setosdisk
+    end
+
+    def premium_os_disk
+    end
+
+    def os_disk_caching
+    end
+
+    def os_disk_caching=(setcaching)
+      @azure_vm.os_disk.caching = setcaching
+    end
+
+    def puppet_master
```

Comment:
This is a bit strange property to have on compute instance. Are you sure it is used by foreman?

Reply:
We do have it here in the VM form: [Puppet Master field.](https://github.com/theforeman/foreman_azure_rm/blob/master/app/views/compute_resources_vms/form/azurerm/_base.html.erb#L177)